src/oData/jsonToOData.ts

```diff
@@ -0,0 +1,175 @@
+		for (const jsonSchemaProp of jsonSchemaProps) {
+			if (value[jsonSchemaProp] != null) {
+				filters.push({ [$op]: value[jsonSchemaProp] });
+			}
+		}
+	}
+	if (filters.length > 0) {
+		return maybePluralOp('$and', filters);
+	}
+	throw new Error('');
+};
+
+const handleFilterArray = (parentKey: string, filterObj: JSONSchema): any => {
+	const filters: any[] = [];
+	const $alias = createAlias(parentKey);
+	if (filterObj.minItems != null && filterObj.minItems > 1) {
+		filters.push({ $ge: [{ [$alias]: { $count: {} } }, filterObj.minItems] });
+	}
+	// Post-MVP: maxItems
+	if (filterObj.contains) {
+		const nestedFilters = filterBuilder([$alias], filterObj.contains);
+		if (nestedFilters) {
+			filters.push(nestedFilters);
+		}
+	}
+
+	if (filters.length === 0) {
+		filters.push({ 1: 1 });
+	}
+	return {
+		[parentKey]: {
+			$any: { $alias, $expr: { [$alias]: maybePluralOp('$and', filters) } },
+		},
+	};
+};
+
+const handleOperators = (parentKeys: string[], filter: JSONSchema): any => {
+	if (!filter.anyOf && !filter.allOf) {
+		throw new Error('Calling handleOperators without anyOf and allOf');
+	}
+	const operator = filter.anyOf || filter.oneOf ? '$or' : '$and';
+	const filters = filter.anyOf || filter.oneOf || filter.allOf;
+	return filterBuilder(parentKeys, maybePluralOp(operator, filters!));
+};
+
+export const filterBuilder = (
+	parentKeys: string[],
+	filter: FilterMutation | FilterMutation[],
+): any => {
+	if (!filter) {
+		return;
+	}
+
+	if (Array.isArray(filter)) {
+		return filter.length > 1
+			? { $and: filter.map((f) => filterBuilder(parentKeys, f)) }
+			: filterBuilder(parentKeys, filter[0]);
+	}
+
+	// TODO: Check if possible to remove and improve
+	if (filter.$or || filter.$and) {
+		const operator = filter.$or ? '$or' : '$and';
+		const filters = filter.$or || filter.$and;
+		return { [operator]: filters?.map((f) => filterBuilder(parentKeys, f)) };
+	}
+
+	if (filter.anyOf || filter.oneOf || filter.allOf) {
+		return handleOperators(parentKeys, filter);
+	}
+
+	if (filter.contains != null || filter.items != null) {
+		return handleFilterArray(parentKeys[parentKeys.length - 1], filter);
+	}
+
+	if (filter.properties != null) {
+		const propFilters: any = Object.entries(filter.properties).map(
+			([key, value]: [string, JSONSchema]) => {
+				return filterBuilder([...parentKeys, key], value);
+			},
+		);
+
+		return maybePluralOp('$and', propFilters);
+	}
+
+	return handlePrimitiveFilter(parentKeys[parentKeys.length - 1], filter);
+};
+
+export const orderbyBuilder = <T>(sortInfo: TableSortOptions<T> | null) => {
+	if (!sortInfo) {
+		return null;
+	}
+	const { field, reverse, refScheme } = sortInfo;
+	if (!field) {
+		return null;
+	}
+	const direction = !reverse ? 'desc' : 'asc';
+	if (!refScheme) {
+		return { [field]: direction };
+	}
+	const converted = refScheme.replace(/\[(.*?)\]/g, '').replace(/\./g, '/');
+	return { [`${field}/${converted}`]: direction };
```

Comment:
NIT: How about v, so that we only have a single exit point that creates the order by object.
```suggestion
	let fieldPath = field;
	if (refScheme) {
		fieldPath += `/${refScheme.replace(/\[(.*?)\]/g, '').replace(/\./g, '/')}`
	}
	return { [fieldPath]: direction };
```